source/extensions/transport_sockets/proxy_protocol/proxy_protocol.cc

```diff
@@ -91,9 +103,21 @@ void UpstreamProxyProtocolSocket::generateHeaderV2() {
   if (!options_ || !options_->proxyProtocolOptions().has_value()) {
     Common::ProxyProtocol::generateV2LocalHeader(header_buffer_);
   } else {
+    // Process any custom TLVs from the host metadata.
+    processCustomTLVsFromHost();
```

Comment:
This will be clearer, and use slightly less memory, if you make this function return the values instead of it being a member variable.

```suggestion
    std::vector<Envoy::Network::ProxyProtocolTLV> custom_tlvs = processCustomTLVsFromHost();
```